### PR TITLE
Summer2020 :QR-code decoder from Zheng Qiushi

### PR DESCRIPTION
**Pull request by ZhengQiushi**
Summer2020 :QR-code decoder from Zheng Qiushi

```
force_builders=Docs,linux,windows,Mac
```

### CODE REVIEW

File: 3rdparty/quirc/src/quirc.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
+/* quirc -- QR-code recognition library
+ * Copyright (C) 2010-2012 Daniel Beer <dlbeer@gmail.com>
```
Comment (dkurt) [inline 3rdparty/quirc/src/quirc.cpp:2]: What is a source of this file? Is that a part of Quirc library or your code?

Am I right that project is dedicated to replacing quirc to OpenCV implementation of QR codes decoding?

**Comment by vpisarev** [comment]
@ZhengQiushi,

1. I agree with @dkurt, please, do not put quirc into OpenCV. OpenCV already uses quirc for decoding.
2. Please, do not put your reports into pull request. Send it to your mentors

**Comment by alalek** [comment]
> from unknown repository

This mistake is not recoverable (do NOT remove forks)

**Comment by dkurt** [comment]
see https://github.com/opencv/opencv/pull/17845

**Pull request closed without merging**